admiral/envs/components/actor.py

```diff
@@ -0,0 +1,248 @@
+                    continue
+                else:
+                    return attacked_agent
+
+
+
+# ----------------------------- #
+# --- Position and Movement --- #
+# ----------------------------- #
+
+class GridMovementActor:
+    """
+    Provides the necessary action space for agents who can move and processes such
+    movements.
+
+    position (PositionState):
+        The position state handler. Needed to modify the agents' positions.
+
+    agents (dict):
+        The dictionary of agents.
+    """
+    def __init__(self, position=None, agents=None, **kwargs):
+        self.position = position
+        self.agents = agents
+
+        from gym.spaces import Box
+        for agent in self.agents.values():
+            if isinstance(agent, GridMovementAgent):
+                agent.action_space['move'] = Box(-agent.move_range, agent.move_range, (2,), np.int)
+
+    def process_move(self, moving_agent, move, **kwargs):
+        """
+        Determine the agent's new position based on its move action.
+
+        moving_agent (GridMovementAgent):
+            The agent that moves.
+        
+        move (np.array):
+            How much the agent would like to move in row and column.
+        
+        return (np.array):
+            How much the agent has moved in row and column. This can be different
+            from the desired move if the position update was invalid.
+        """
+        if isinstance(moving_agent, GridMovementAgent) and isinstance(moving_agent, PositionAgent):
+            position_before = moving_agent.position
+            self.position.modify_position(moving_agent, move, **kwargs)
+            return moving_agent.position - position_before
+
+class SpeedAngleMovementActor:
+    """
+    Process acceleration and angle changes for SpeedAngleAgents. Update the agents'
+    positions based on their new speed and direction.
+
+    position (ContinuousPositionState):
+        The position state handler. Needed to modify agent positions.
+    
+    speed_angle (SpeedAngleState):
+        The speed and angle state handler.
+
+    agents (dict):
+        The dictionary of agents.
+    """
+    def __init__(self, position=None, speed_angle=None, agents=None, **kwargs):
+        self.position = position
+        self.speed_angle = speed_angle
+        self.agents = agents
+
+        from gym.spaces import Box
+        for agent in agents.values():
+            if isinstance(agent, SpeedAngleAgent):
+                agent.action_space['accelerate'] = Box(-agent.max_acceleration, agent.max_acceleration, (1,))
+                agent.action_space['banking_angle_change'] = Box(-agent.max_banking_angle_change, agent.max_banking_angle_change, (1,))
+    
+    def move(self, agent, acceleration, angle, **kwargs):
+        self.speed_angle.modify_speed(agent, acceleration)
+        self.speed_angle.modify_banking_angle(agent, angle)
+        
+        x_position = agent.speed*np.cos(np.deg2rad(agent.ground_angle))
```

Comment:
In GridPositions, the first element corresponds to the row and the second corresponds to the column, which results in something like (y, x). We should keep this (y, x) ordering the same, so either change the ordering for Continuous for for Grid, keep in mind that we had it like (x,y) for Grid before but changed it for some reasons.

Reply:
No, it's okay. It's a different spatial representation. Just leave a note about it in the state or agent.